src/Compilers/CSharp/Portable/Symbols/Source/SourceFieldSymbol.cs

```diff
@@ -390,5 +395,17 @@ private ConstantValue GetLazyConstantValue(bool earlyDecodingWellKnownAttributes
         }
 
         protected abstract ConstantValue MakeConstantValue(HashSet<SourceFieldSymbolWithSyntaxReference> dependencies, bool earlyDecodingWellKnownAttributes, BindingDiagnosticBag diagnostics);
+
+        internal override void PostDecodeWellKnownAttributes(ImmutableArray<CSharpAttributeData> boundAttributes, ImmutableArray<AttributeSyntax> allAttributeSyntaxNodes, BindingDiagnosticBag diagnostics, AttributeLocation symbolPart, WellKnownAttributeData decodedData)
+        {
+            if (this.state.HasComplete(CompletionPart.ConstantValue))
+            {
+                CheckConstantValue(_lazyConstantValue, diagnostics);
+            }
+
+            base.PostDecodeWellKnownAttributes(boundAttributes, allAttributeSyntaxNodes, diagnostics, symbolPart, decodedData);
+        }
+
+        protected virtual void CheckConstantValue(ConstantValue value, BindingDiagnosticBag diagnostics) { }
```

Comment:
> protected virtual void CheckConstantValue(ConstantValue value, BindingDiagnosticBag diagnostics) { } [](http://example.com/codeflow?start=7&length=101)

Adding virtual methods to symbols is error prone. It is easy to fail to override it when appropriate. Our general approach is to add abstract methods and provide implementation for most derived types only. Alternatively, we might find a way to move this check to SourceMemberFieldSymbol without introducing new APIs